bin/run-functional-tests.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# Defaults
+: ${PYTEST_PROCESSES:="1"}
+: ${BASE_URL:="https://www-dev.allizom.org"}
+: ${DRIVER:="Remote"}
+: ${SELENIUM_PORT:="4444"}
+: ${BROWSER_NAME:="firefox"}
+: ${TESTS_PATH:="tests/functional"}
+: ${RESULTS_PATH:="${TESTS_PATH}/results"}
+
+# Common arguments
+CMD="py.test"
+CMD="${CMD} -n ${PYTEST_PROCESSES}"
+CMD="${CMD} --base-url ${BASE_URL}"
+CMD="${CMD} --driver ${DRIVER}"
+CMD="${CMD} --html ${RESULTS_PATH}/index.html"
+CMD="${CMD} --junitxml ${RESULTS_PATH}/junit.xml"
+
+# Remote arguments
+if [ "${DRIVER}" = "Remote" ]; then
+  CMD="${CMD} --host ${SELENIUM_HOST}"
+  CMD="${CMD} --port ${SELENIUM_PORT}"
+  CMD="${CMD} --capability browserName \"${BROWSER_NAME}\""
+  if [ -n "${BROWSER_VERSION}" ]; then CMD="${CMD} --capability version \"${BROWSER_VERSION}\""; fi
+  if [ -n "${PLATFORM}" ]; then CMD="${CMD} --capability platform \"${PLATFORM}\""; fi
+fi
+
+# Sauce Labs arguments
+if [ "${DRIVER}" = "SauceLabs" ]; then
+  CMD="${CMD} --capability browserName \"${BROWSER_NAME}\""
+  if [ -n "${BROWSER_VERSION}" ]; then CMD="${CMD} --capability version \"${BROWSER_VERSION}\""; fi
+  if [ -n "${PLATFORM}" ]; then CMD="${CMD} --capability platform \"${PLATFORM}\""; fi
+  if [ -n "${SELENIUM_VERSION}" ]; then CMD="${CMD} --capability selenium-version \"${SELENIUM_VERSION}\""; fi
+  if [ -n "${BUILD_TAG}" ]; then CMD="${CMD} --capability build \"${BUILD_TAG}\""; fi
+  if [ -n "${SCREEN_RESOLUTION}" ]; then CMD="${CMD} --capability screenResolution \"${SCREEN_RESOLUTION}\""; fi
+  if [ -n "${PRIVACY}" ]; then CMD="${CMD} --capability public \"${PRIVACY}\""; fi
+fi
+
+CMD="${CMD} ${TESTS_PATH}"
+echo "Running: ${CMD}" && eval ${CMD}
```

Comment:
I like to use `set -xe` at the start of my bash scripts. `-x` prints the commands before executing them, similarly to what you do here with `Running: ${CMD}` and `-e` exits immediatelly after a failed cmd.